README.md

```diff
@@ -1,51 +1,75 @@
+*   **Simplify Ingress:** Manage TCP/UDP forwarding declaratively using Kubernetes custom resources.
+*   **Achieve High Availability:** Run multiple replicas of the proxy for resilience.
+
+Each `UnDyingProxy` object specifies one listen port and one target host and port to forward to each for TCP and UDP.
+
+This supports only paying for one Loadbalancer, or even avoiding a cloud Loadbalancer altogether by running as a DaemonSet on nodes with direct public IPs. If using a cloud Loadbalancer, this operator can automatically add/remove `Ports` in a `Service` object as `UnDyingProxy` objects are created/destroyed, to open/close the ports in the Loadbalancer.
+
+## ✨ Features
+
+*   **TCP & UDP Forwarding:** Configure forwarding rules for both protocols independently per `UnDyingProxy` resource.
+*   **Dynamic Configuration:** Define forwarding rules using the `UnDyingProxy` Custom Resource Definition (CRD). Changes are automatically detected and applied.
+*   **Automatic Service Port Management (Optional):** Automatically update Kubernetes `Service` ports (e.g., for LoadBalancer Services) as `UnDyingProxy` resources are created/deleted.
+*   **High Availability:** Designed to run with multiple replicas behind a Kubernetes Service; leader election is disabled.
+*   **Prometheus Metrics:** Exposes detailed metrics for monitoring traffic and operator health.
+*   **Configurable Timeouts & Buffers:** Fine-tune UDP connection handling.
 
-Cloud load balancers can be expensive. This operator forwards TCP and UDP traffic from a single IP and many ports, to many destinations, one destination per source port for now. It is designed to be run as multiple replicas to provide high availability.
+## Example Use Case
 
-Each `UnDyingProxy` object specifies one listen port and one target host and port to forward to each for TCP and UDP. e.g.
+Forward external UDP port `27015` to an internal game server and external TCP port `5432` to a database, all managed by the same UnDyingProxy deployment listening on a single IP.
 
 ```yaml
 ---
 apiVersion: proxy.sfact.io/v1alpha1
 kind: UnDyingProxy
 metadata:
-  name: example
+  name: game-server-1001
+  namespace: undying-proxy # Must match operator's namespace
+spec:
+  udp:
+    listenPort: 27015 # External port on the proxy's IP
+    targetPort: 27015 # Internal port of the game server pod/service
+    targetHost: my-game-server-1001.game-ns.svc.cluster.local
+---
+apiVersion: proxy.sfact.io/v1alpha1
+kind: UnDyingProxy
+metadata:
+  name: database-tunnel
   namespace: undying-proxy
-  labels:
-    app: undying-proxy
 spec:
   tcp:
-    listenPort: 1234
-    targetPort: 1234
-    targetHost: my-app.example.svc.cluster.local
-  udp:
-    listenPort: 1234
-    targetPort: 1234
-    targetHost: my-app.example.svc.cluster.local
+    listenPort: 5432 # External port
+    targetPort: 5432 # Internal port
+    targetHost: my-postgres-service.db-ns.svc.cluster.local
 ```
 
```

Comment:
UnDyingProxy can also be run as a DaemonSet along with round robin DNS to point a domain to your node IPs, and potentially avoid a cloud Loadbalancer altogether. This requires clients to connect directly to node IPs.